### ISSUE TITLE
Fetch from Tasklist URL: embedded String

### ISSUE THREAD

**Issue by MaxTru** [camunda/connectors#1432]
__Describe the Bug__

I am trying to fetch a task from Tasklist REST API using the REST POST Connector.

The post request works using CURL just fine:
```
curl -X 'POST' \
  'https://bru-2.tasklist.camunda.io/8da2f67e-88ab-4b38-9e39-52bb0ea4855f/v1/tasks/search' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -H "Authorization: Bearer <omitted>" \
  -d '{
      "state": "CREATED",
      "taskVariables": [
        {
          "name": "epicURL",
          "value": "\"https://github.com/camunda/product-hub/issues/1905\"",
          "operator": "eq"
        }
      ]
    }'
```

=> This works

Using the REST Post Connector, the Tasklist API is returning a 200, **but 0 tasks**. The problem is the `taskVariables` filter (if I omit it, then I get tasks returned). I tried the following variations:
<img width="397" alt="Bildschirmfoto 2023-11-17 um 10 55 02" src="https://github.com/camunda/connectors/assets/42800119/133d5182-4df5-4574-bb98-ff4c0a9562aa">

Variant 1: (request body)
```
{
  state: "CREATED",
  taskVariables: [
    {
      name: "epicURL",
      value: "\"https://github.com/camunda/product-hub/issues/1905\"",
      operator: "eq"
    }
  ]
}
```

Variant 2: (request body)
```
{
  state: "CREATED",
  taskVariables: [
    {
      name: "epicURL",
      value: "https://github.com/camunda/product-hub/issues/1905",
      operator: "eq"
    }
  ]
}
```

__Steps to Reproduce__

1. Use REST POST Connector
2. Try to fetch something from Tasklist URL using the TaskVariables filter, filtering for a string value https://docs.camunda.io/docs/next/apis-tools/tasklist-api-rest/schemas/requests/task-search-request/#
3. While this works in CURL, I get 0 tasks returned via the POST Connector

__Expected Behavior__

I can make my CURL POST request work in the REST Connector

__Environment__

8.3.2 SaaS Prod Cluster

**Comment by MaxTru** [comment]
Related to https://github.com/camunda/tasklist/issues/3754

**Comment by MaxTru** [comment]
Basically:
I can get this to work with CURL, but not with REST POST Connector. I don't have a way to "debug" what the REST Connector is doing here.

**Comment by sbuettner** [comment]
@MaxTru Looks like a duplicate of https://github.com/camunda/connectors/issues/239#issuecomment-1434850976 WDYT?

**Comment by MaxTru** [comment]
I investigated again.

So I provided to the Connector (via prop panel) the following:
```json
{
  state: "CREATED",
  taskVariables: [
    {
      name: "epicURL",
      value: "\"https://github.com/camunda/product-hub/issues/1905\"",
      operator: "eq"
    }
  ]
}
```

This is what the REST endpoint retrieves:
```json
{"state":"CREATED","taskVariables":[{"name":"epicURL","value":"\\\"https://github.com/camunda/product-hub/issues/1905\\\"","operator":"eq"}]}
```

(Formatted)
```json
{
  "state":"CREATED",
  "taskVariables": [
     {
       "name":"epicURL",
        "value":"\\\"https://github.com/camunda/product-hub/issues/1905\\\"",
        "operator":"eq"
      }
    ]
}
```

**Comment by MaxTru** [comment]
> @MaxTru Looks like a duplicate of [#239 (comment)](https://github.com/camunda/connectors/issues/239#issuecomment-1434850976) WDYT?

I agree Simon.

**Comment by MaxTru** [comment]
Tested on a `Zeebe 8.3.4 cluster`on December 15:

* **Test 1** Sent this request via REST POST Connector
  * Payload as defined in the properties panel:
```
{
  state: "CREATED",
  taskVariables: [
    {
      name: "epicURL",
      value: "\"https://github.com/camunda/product-hub/issues/1905\"",
      operator: "eq"
    }
  ]
}
```
  * This is what the endpoint retrieves:
```
{
  "state":"CREATED",
  "taskVariables":
    [
       {"name":"epicURL",
       "value":"\\\"https://github.com/camunda/product-hub/issues/1905\\\"",
       "operator":"eq"}
    ]
}
```

* **Test 2** Sent this request via REST POST Connector
  * Payload as defined in the properties panel:
```
{
  state: "CREATED",
  taskVariables: [
    {
      name: "epicURL",
      value: "https://github.com/camunda/product-hub/issues/1905",
      operator: "eq"
    }
  ]
}
```
  * This is what the endpoint retrieves:
```
{
  "state":"CREATED",
  "taskVariables":
    [
       {"name":"epicURL",
       "value":"https://github.com/camunda/product-hub/issues/1905",
       "operator":"eq"}
    ]
}
```


----------

My take-away:
* The behavior has improved with the Zeebe fix (see test case 2)
* However, Connectors is still not useable in conjunction with Tasklist API, since Tasklist API expects `\", which is still double-escaped (see test case 1).

@sbuettner with the Zeebe side now (apparently?) fixed ([ref](https://github.com/camunda/feel-scala/pull/750)), what is the next step?

**Comment by sbuettner** [comment]
Hi @MaxTru,
thanks for testing. 

Test 1: Do you know how the value looked in Operate when the connector received it? 

Maybe @saig0 can provide an answer whether the first case should be covered with the latest FEEL engine fix? 
 
Regarding Test 2: Maybe I am missing something but based on the payload you provided I cant see how it should be related to escaping.

**Comment by MaxTru** [comment]
> Test 1: Do you know how the value looked in Operate when the connector received it?

Sure - see screenshot:

![Bildschirmfoto 2023-12-18 um 15 22 22](https://github.com/camunda/connectors/assets/42800119/8b456f6f-2bf6-4556-a67c-c734b041125a)


> Regarding Test 2: Maybe I am missing something but based on the payload you provided I cant see how it should be related to escaping.

I just wanted to test that intendation works - you are right, this has nothing to do with escaping per se.

**Comment by sbuettner** [comment]
Thanks, as it arrived like this in Operate it looks like its not covered by the escaping fix as this is the data that will be handed over to the Connectors runtime and the Connector cant to much about it. 
@saig0 Looks like there is still some escaping taking place and maybe its related how expression results are handed over to the FEEL engine by zeebe: https://github.com/camunda/feel-scala/issues/701 cc @nicpuppa

**Comment by saig0** [comment]
@sbuettner true. We still have an issue with double quotes: https://github.com/camunda/feel-scala/issues/778
